### PR TITLE
Account event

### PR DESCRIPTION
**Pull request by dholms**
Implementation for schemas in https://github.com/bluesky-social/atproto/pull/2263

### CODE REVIEW

File: packages/bsky/src/data-plane/server/subscription/index.ts [hunk 1]
```diff
@@ -270,6 +276,8 @@ function getMessageDetails(msg: Message):
     return { seq: msg.seq, repo: msg.did, message: msg }
   } else if (message.isIdentity(msg)) {
     return { seq: msg.seq, repo: msg.did, message: msg }
+  } else if (message.isAccount(msg)) {
+    return { seq: msg.seq, repo: msg.did, message: msg }
```
Comment (devinivy) [inline packages/bsky/src/data-plane/server/subscription/index.ts:280]: I think there's one more of these next door in `subscription/util.ts` within the implementation for `loggableMessage()`!

**Review by devinivy — APPROVED** [review]
Ace!

File: packages/bsky/src/data-plane/server/subscription/index.ts (hunk 1 above)
Comment (dholms) [reply packages/bsky/src/data-plane/server/subscription/index.ts:280]: ah good snag 👍

File: packages/pds/src/account-manager/index.ts [hunk 2]
```diff
@@ -71,6 +68,22 @@ export class AccountManager {
     return got?.did ?? null
   }
 
+  async getAccountStatus(handleOrDid: string): Promise<AccountStatus> {
+    const got = await this.getAccount(handleOrDid, {
+      includeDeactivated: true,
+      includeTakenDown: true,
+    })
+    if (!got) {
+      return AccountStatus.Deactivated
```
Comment (devinivy) [inline packages/pds/src/account-manager/index.ts:77]: Is this one intended to be `AccountStatus.Deleted`?

File: packages/pds/src/api/com/atproto/identity/submitPlcOperation.ts [hunk 3]
```diff
@@ -47,14 +46,6 @@ export default function (server: Server, ctx: AppContext) {
...
       await ctx.plcClient.sendOperation(requester, op)
       await ctx.sequencer.sequenceIdentityEvt(requester)
-      try {
-        await ctx.idResolver.did.resolve(requester, true)
-      } catch (err) {
-        log.error(
-          { err, did: requester },
-          'failed to refresh did after plc update',
-        )
-      }
```
Comment (devinivy) [inline packages/pds/src/api/com/atproto/identity/submitPlcOperation.ts:57]: Was this unnecessary, or did the functionality move somewhere?

File: packages/pds/src/api/com/atproto/server/util.ts [hunk 4]
```diff
@@ -28,21 +28,29 @@ export const getRandomToken = () => {
...
   try {
     const didDoc = await ctx.idResolver.did.resolve(did, forceRefresh)
     return didDoc ?? undefined
   } catch (err) {
-    dbLogger.warn({ err, did }, 'failed to resolve did doc')
+    httpLogger.warn({ err, did }, 'failed to resolve did doc')
   }
 }
 
+// @TODO once supporting multiple pdses, validate pds in did doc based on allow-list.
```
Comment (devinivy) [inline packages/pds/src/api/com/atproto/server/util.ts:44]: Not sure if I understand this TODO anymore 🤔 could it have been meant for entryway?

File: packages/pds/src/api/com/atproto/sync/getRepoStatus.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      const active = !account.takedownRef && !account.deactivatedAt
+
+      let status: string | undefined = undefined
+      if (!active) {
+        if (account.takedownRef) {
+          status = AccountStatus.Takendown
+        } else if (account.deactivatedAt) {
+          status = AccountStatus.Deactivated
+        }
+      }
```
Comment (devinivy) [inline packages/pds/src/api/com/atproto/sync/getRepoStatus.ts]: I would almost flip these around:
```suggestion
      let status: string | undefined = undefined
      if (account.takedownRef) {
        status = AccountStatus.Takendown
      } else if (account.deactivatedAt) {
        status = AccountStatus.Deactivated
      }

     const active = !status
```

File: packages/pds/tests/sync/subscribe-repos.test.ts [hunk 6]
```diff
@@ -323,6 +372,64 @@ describe('repo subscribe repos', () => {
...
+      {
+        encoding: 'application/json',
+        headers: sc.getHeaders(alice),
+      },
+    )
+    await agent.api.com.atproto.server.activateAccount(undefined, {
+      headers: sc.getHeaders(alice),
+    })
+
+    // takedown then restore bob
```
Comment (devinivy) [inline packages/pds/tests/sync/subscribe-repos.test.ts:388]: I _think_ that it's covered alright, but what about flows where alice is deactivated while takendown, or is takendown while deactivated?  (Or activated while takendown?  I think I see this one at least would error.)

Would they look right both to a firehose consumer and on `getRepoStatus`?

**Review by devinivy — APPROVED** [review]
Looks great 👍

File: packages/pds/src/account-manager/index.ts (hunk 2 above)
Comment (dholms) [reply packages/pds/src/account-manager/index.ts:77]: ah yup good snag 👍

File: packages/pds/src/api/com/atproto/server/util.ts (hunk 4 above)
Comment (dholms) [reply packages/pds/src/api/com/atproto/server/util.ts:44]: yeah p sure it's meant for entryway 👍

File: packages/pds/src/api/com/atproto/sync/getRepoStatus.ts (hunk 5 above)
Comment (dholms) [reply packages/pds/src/api/com/atproto/sync/getRepoStatus.ts]: yeah i dig 👍

File: packages/pds/tests/sync/subscribe-repos.test.ts (hunk 6 above)
Comment (dholms) [reply packages/pds/tests/sync/subscribe-repos.test.ts:388]: good call, probably worth a test 👍

**Pull request merged**